webapp/utils/utils.jsx

```diff
@@ -1372,3 +1372,47 @@ export function localizeMessage(id, defaultMessage) {
 export function mod(a, b) {
     return ((a % b) + b) % b;
 }
+
+export function isValidPassword(password, requirements) {
+    let errorMsg = '';
+    let errorId = 'user.settings.security.passwordError';
+    let error = false;
+
+    if (password.length < requirements.MinimumLength || password.length > Constants.MAX_PASSWORD_LENGTH) {
+        error = true;
+    }
+
+    if (requirements.Lowercase && !password.match(/[a-z]/)) {
+        errorId = errorId + 'Lowercase';
+        error = true;
+    }
+
+    if (requirements.Number && !password.match(/[0-9]/)) {
+        errorId = errorId + 'Uppercase';
+        error = true;
+    }
+
+    if (requirements.Uppercase && !password.match(/[A-Z]/)) {
+        errorId = errorId + 'Number';
+        error = true;
+    }
+
+    if (requirements.Symbol && !password.match(/[ !"#$%&'()*+,-./:;<=>?@[\]^_`|~]/)) {
```

Comment:
You need to escape the hyphen since it has special meaning in square brackets. `/[ !"#$%&'()*+,\-./:;<=>?@[\]^_`|~]/`